components/feature/autofill/src/main/java/mozilla/components/feature/autofill/handler/FillRequestHandler.kt

```diff
@@ -43,24 +46,70 @@ internal class FillRequestHandler(
      * Handles a fill request for the given [AssistStructure] and returns a matching [FillResponse]
      * or `null` if the request could not be handled or the passed in [AssistStructure] is `null`.
      */
+    @SuppressLint("InlinedApi")
     @Suppress("ReturnCount")
-    suspend fun handle(structure: AssistStructure?, forceUnlock: Boolean = false): FillResponse? {
+    suspend fun handle(
+        structure: AssistStructure?,
+        forceUnlock: Boolean = false
+    ): FillResponse? {
         if (structure == null) {
             return null
         }
 
         val parsedStructure = parseStructure(context, structure) ?: return null
-        val logins = configuration.storage.getByStructure(configuration.publicSuffixList, parsedStructure)
+        val lookupDomain = parsedStructure.getLookupDomain(configuration.publicSuffixList)
+        val needsConfirmation = !configuration.verifier.hasCredentialRelationship(
```

Comment:
Funny thing, from brief research some stores appear to strip original package signatures (e.g. amazon's app store may) and re-sign themselves. That would obviously invalidate the check. Not that it should affect us here, just figured i'd mention it.

Reply:
Yeah, in this case the third-party would have to add multiple entries to their digital assets links file.